### PR TITLE
Add test for batch decoding

### PR DESCRIPTION
**Pull request by LEAFERx**
# Core changes
- Added a unit-test for new format batch decoding with l1 oracle transaction

### CODE REVIEW

File: src/specular/stages/batches.rs [hunk 1]
```diff
@@ -308,3 +308,189 @@ fn try_decode_l1_oracle_values(
...
+            let state_root: H256 = Default::default();
+            let first_l2_block_num = 1;
+            let l1_inclusion_block = 1;
+
+            let fake_signature = Signature {
+                v: 0,
+                r: Default::default(),
+                s: Default::default(),
+            };
+            let oralce_tx_values: SetL1OracleValuesInput = (
```
Comment (chalkandpaste) [inline src/specular/stages/batches.rs]: `s/oralce_tx_values/oracle_tx_values/`

File: src/specular/stages/batches.rs [hunk 2]
```diff
@@ -308,3 +308,189 @@ fn try_decode_l1_oracle_values(
...
+                epoch_num.into(),
+                timestamp.into(),
+                base_fee.into(),
+                epoch_hash,
+                state_root,
+            );
+            let oracle_tx_data = SET_L1_ORACLE_VALUES_ABI
+                .encode_with_selector(*SET_L1_ORACLE_VALUES_SELECTOR, oralce_tx_values)?;
+            let oralce_tx: TypedTransaction = TransactionRequest::new()
```
Comment (chalkandpaste) [inline src/specular/stages/batches.rs]: `s/oralce_tx/oracle_tx/`

**Review by chalkandpaste — APPROVED** [review]
Aside from typos, LGTM

File: src/specular/stages/batches.rs (hunk 1 above)
Comment (LEAFERx) [reply src/specular/stages/batches.rs]: fixed, nice catch

File: src/specular/stages/batches.rs (hunk 2 above)
Comment (LEAFERx) [reply src/specular/stages/batches.rs]: fixed

**Review by chalkandpaste — APPROVED** [review]

**Review by ujvl — APPROVED** [review]

**Pull request merged**